Source/src/importers/AnimationImporter.cpp

```diff
@@ -7,33 +7,62 @@
 
 Hachiko::AnimationImporter::AnimationImporter() : Importer(Importer::Type::ANIMATION) {}
 
-void Hachiko::AnimationImporter::Import(const char* path) {}
+void Hachiko::AnimationImporter::Import(const char* path) {
 
-void Hachiko::AnimationImporter::Save(const Resource* resource) {}
+    HE_LOG("Entering Animation Importer: %s", path);
+    Assimp::Importer import;
 
-Hachiko::Resource* Hachiko::AnimationImporter::Load(const char* model_path)
-{
-    return nullptr;
+    const aiScene* scene = nullptr;
+    const std::filesystem::path material_path(path);
+
+    scene = import.ReadFile(path, aiProcess_Triangulate | aiProcess_FlipUVs | aiProcess_CalcTangentSpace);
+
+    assert(scene->mNumAnimations == 1);
+
+    Import(scene->mAnimations[0]);
+}
+
+void Hachiko::AnimationImporter::Save(const Resource* resource) {
+
+    const ResourceAnimation* animation = static_cast<const ResourceAnimation*>(resource);
+    const std::string animation_library_path = GetResourcesPreferences()->GetAssetsPath(Resource::Type::ANIMATION) + animation->GetName();
+
+    
+    /* TODO: ANIMATION  Save animation as binary like mesh*/
+
+    // FileSystem::Save(animation_library_path.c_str(), animation_node);
 }
 
-bool Hachiko::AnimationImporter::IsImported(const char* path)
+Hachiko::Resource* Hachiko::AnimationImporter::Load(const char* animation_path)
 {
-    return false;
+    if (FileSystem::Exists(animation_path))
+    {
+        return nullptr;
+    }
+
+    char* file_buffer = FileSystem::Load(animation_path);
+
+    std::string animation_id = FileSystem::GetFileNameAndExtension(animation_path);
+
+    const auto animation = new ResourceAnimation(UUID::StringToUID(animation_id));
+
+    /* TODO: ANIMATION add all info to charge the node into resource */
+
```

Comment:
I think this method is going to be completed in the future prs, but in the mean time, to be sure we don't have any memory leaks, let's either delete/comment the line 43 or delete the ```file_buffer``` at the end of the method.
```suggestion
    char* file_buffer = FileSystem::Load(animation_path);

    std::string animation_id = FileSystem::GetFileNameAndExtension(animation_path);

    const auto animation = new ResourceAnimation(UUID::StringToUID(animation_id));
    
    delete[] file_buffer;
    
    /* TODO: ANIMATION add all info to charge the node into resource */

```